idea/idea-completion/src/org/jetbrains/kotlin/idea/completion/handlers/handlerUtils.kt

```diff
@@ -144,8 +146,14 @@ fun createKeywordConstructLookupElement(
                     val beforeCaret = tailBeforeCaret.indentLinesAfterFirst(newIndent)
                     val afterCaret = tailAfterCaret.indentLinesAfterFirst(newIndent)
 
-                    insertionContext.document.insertString(offset, beforeCaret + afterCaret)
-                    insertionContext.editor.moveCaret(offset + beforeCaret.length)
+                    val element = insertionContext.file.findElementAt(offset)?.getNextSiblingIgnoringWhitespaceAndComments(true)
```

Comment:
The following a bit artificial snippet can be used as an example of unexpected behaviour:

```
fun other() {
  try<caret>

  // Some
  {
    println()
  }()
}
```